tests/apps/test_graph_similarity.py

```diff
@@ -89,3 +92,113 @@ def test_sample_to_event(dim, max_count_per_mode):
     events = [similarity.sample_to_event(o, max_count_per_mode) for o in orbits]
 
     assert events == target_events
+
+
+class TestOrbitToSample:
+    """Tests for the function ``strawberryfields.apps.graph.similarity.orbit_to_sample``"""
+
+    def test_low_modes(self):
+        """Test if function raises a ``ValueError`` if fed an argument for ``modes`` that does
+        not exceed the length of the input orbit."""
+        with pytest.raises(ValueError, match="Number of modes cannot"):
+            similarity.orbit_to_sample([1, 2, 3], 2)
+
+    @pytest.mark.parametrize("orb_dim", [3, 4, 5])
+    @pytest.mark.parametrize("modes_dim", [6, 7])
+    def test_sample_length(self, orb_dim, modes_dim):
+        """Test if function returns a sample that is of correct length ``modes_dim`` when fed a
+        collision-free event of ``orb_dim`` photons."""
+        samp = similarity.orbit_to_sample(all_orbits[orb_dim][0], modes_dim)
+        assert len(samp) == modes_dim
+
+    def test_sample_composition(self):
+        """Test if function returns a sample that corresponds to the input orbit. Input orbits
+        are orbits from ``all_orbits_cumulative``, i.e., all orbits from 3-5 photons. This test
+        checks if a sample corresponds to an orbit by counting the occurrence of elements in the
+        sample and comparing to a count of elements in the orbit."""
+        modes = 5
+
+        all_orbits_zeros = [
+            [1, 1, 1, 0, 0],
+            [2, 1, 0, 0, 0],
+            [3, 0, 0, 0, 0],
+            [1, 1, 1, 1, 0],
+            [2, 1, 1, 0, 0],
+            [3, 1, 0, 0, 0],
+            [2, 2, 0, 0, 0],
+            [4, 0, 0, 0, 0],
+            [1, 1, 1, 1, 1],
+            [2, 1, 1, 1, 0],
+            [3, 1, 1, 0, 0],
+            [2, 2, 1, 0, 0],
+            [4, 1, 0, 0, 0],
+            [3, 2, 0, 0, 0],
+            [5, 0, 0, 0, 0],
+        ]  # padding orbits with zeros at the end for comparison to samples
+
+        counts = [Counter(similarity.orbit_to_sample(o, modes)) for o in all_orbits_cumulative]
+        ideal_counts = [Counter(o) for o in all_orbits_zeros]
+
+        assert counts == ideal_counts
+
+
+class TestEventToSample:
+    """Tests for the function ``strawberryfields.apps.graph.similarity.event_to_sample``"""
+
+    def test_low_count(self):
+        """Test if function raises a ``ValueError`` if ``max_count_per_mode`` is not positive."""
+        with pytest.raises(ValueError, match="Maximum number of photons"):
+            similarity.event_to_sample(2, 0, 5)
+
+    def test_high_photon(self):
+        """Test if function raises a ``ValueError`` if ``photon_number`` is so high that it
+        cannot correspond to a sample given the constraints of ``max_count_per_mode`` and
+        ``modes``"""
+        with pytest.raises(ValueError, match="No valid samples can be generated."):
+            similarity.event_to_sample(5, 1, 4)
+
+    @pytest.mark.parametrize("photon_num", [3, 4, 5, 6])
```

Comment:
Does this test loop over 4x2x4 = 32 different input parameters? If so, is it worth decreasing this?

Reply:
tbh, if it takes an insignificant amount of time, testing 32 input parameters should be okay :slightly_smiling_face: 
 The only time it is worth removing parameters you are testing is if it is a time sink when running the tests.

Reply:
It's ok to do more, but I don't think we gain much by doing it, so best to keep things small